beets/ui/commands.py

```diff
@@ -680,10 +680,13 @@ def show_match_tracks(self):
                 # Save new medium details for future comparison.
                 medium, disctitle = track_info.medium, track_info.disctitle
 
-            if config["import"]["detail"]:
-                # Construct the line tuple for the track.
-                left, right = self.make_line(item, track_info)
+            # Construct the line tuple for the track.
+            left, right = self.make_line(item, track_info)
+            if right["contents"] != "":
                 lines.append((left, right))
+            else:
+                if config["import"]["detail"]:
+                    lines.append((left, right))
```

Comment:
Personally I think the way I wrote it is more readable. This way it's "is there a change? If not, do we want to show it anyway?"

Reply:
My 2 cents of newcomer: what is not easily understandable here is `right["contents"] != ""` .

After looking at the code, I understand that right["contents"] refers to the right column generated by make_line.  
What could be more readable would be to move the code after call to make_line to a `filter_track_lines(left, right)` 

Also, a design remark: the code generates the text to to display, then filters it. We could be more parsimonious by moving the `if config["import"]["detail"]`  in the make_line functions.

Reply:
I don't think a new function is needed here, reading through the function it should be very clear what is happening. Just looking at a line in isolation isn't how we should design readability.

As for the generation, it should stay the same imo. Generating the text isn't expensive and before it's generated, there's no way to know whether or not we need it, since the changes are always displayed and others are dependent on the configuration.

Reply:
> My 2 cents of newcomer: what is not easily understandable here is `right["contents"] != ""` .

This was probably the reason why I initially bothered to suplly a suggestion. In the end I didn't find a way to make this line better (The left != right thing read nicely and was kind of clear without context). I thougth about making this != to a plain ==, but also that failed and didn't add to less confusion.

So in the end I had the feeling that a one-line version using the "or", as I supplied makes the whole thing easier to grasp in one glance.

But anyway, I'm also very happy with what @Serene-Arc thinks about the initial version and I think we can happlily merge this essential fix :-)

Reply:
Yay :)